pinot-spi/src/main/java/org/apache/pinot/spi/trace/InvocationRecording.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.trace;
+
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+public interface InvocationRecording {
```

Comment:
(nit) brief javadocs please

Also, IIUC the granularity of this is at each operator level ? In that case, may be we should call it OperatorInvocationRecording ?

Reply:
I changed the name because I realised to be able to trace the right events we need to move beyond operators and trace the blocks themselves, so I chose a generic name like `Invocation`. I'm also trying to avoid terms common in distributed tracing because the motivation here is contextual profiling, not correlation across components.